azkaban-common/src/main/java/azkaban/utils/DependencyTransferManager.java

```diff
@@ -61,6 +66,7 @@ public DependencyTransferManager(final Props props, final Storage storage) {
         new ThreadFactoryBuilder().setNameFormat("azk-dependency-pool-%d").build());
     this.dependencyMaxDownloadTries =
         props.getInt(Constants.ConfigurationKeys.AZKABAN_DEPENDENCY_MAX_DOWNLOAD_TRIES, 2);
+    this.dependencyDownloadMaxTimeout = props.getInt(AZKABAN_DEPENDENCY_DOWNLOAD_TIMEOUT_SECONDS, 300);
```

Comment:
Do you have some metrics to show if 5min is enough or not?

Reply:
unfortunately I don't know if that's enough, I am using your log "time takes to download" as reference, it's usually really fast. most of them finished with 0 or 1s. 
But again I am uncertain about that. So in case any issues we can quickly restart the webserver by changing config.